File: saas_app/lib/features/products/domain/entities/product.dart
class Product {
  final int id;
  final String name;
  final String description;
  final double price;
  final int quantity;
  final String categoryName;
  final int categoryId;
  final String unit;
  final int supplierId;
  final String? imageUrl;
  final double? rating;
  final DateTime? createdAt;

  const Product({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.quantity,
    required this.categoryName,
    required this.categoryId,
    required this.unit,
    required this.supplierId,
    this.imageUrl,
    this.rating,
    this.createdAt,
  });
}

File: saas_backend/app/Models/Product.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class Product extends Model
{
    use SoftDeletes;

    protected $fillable = [
        'supplier_id',
        'category_id',
        'name',
        'unit',
        'price',
        'min_order_qty',
        'stock',
        'description',
        'images',
    ];

    protected $casts = [
        'images' => 'array',
        'price' => 'decimal:2',
    ];

    /**
     * Get the supplier (company) that owns this product.
     */
    public function supplier()
    {
        return $this->belongsTo(Company::class, 'supplier_id');
    }

    /**
     * Get the category that owns this product.
     */
    public function category()
    {
        return $this->belongsTo(Category::class);
    }

    /**
     * Get all order items for this product.
     */
    public function orderItems()
    {
        return $this->hasMany(OrderItem::class);
    }
}
